src/components/types/Number.js

```diff
@@ -18,16 +18,30 @@ class NumberType extends React.Component {
   render() {
     const { knob, onChange } = this.props;
 
-    return (
-      <input
-        id={knob.name}
-        ref="input"
-        style={styles}
-        value={knob.value}
-        type="number"
-        onChange={() => onChange(parseFloat(this.refs.input.value))}
-      />
-    );
+	  if (knob.range) {
+		  return (
```

Comment:
create two methods returning these two types of input elements.
Then use a turnery to render these like:

````js
return knob.range ?  this.renderNormal() : this.randerRange();
````